### PR TITLE
Add pycudadecon 

### PR DESCRIPTION
**Pull request by tlambert03**
closes #15059

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by tlambert03** [comment]
@zbarry ... looks like this one is going to pass.  Can you leave a comment here stating that you're willing to be listed as a maintainer?

**Comment by zbarry** [comment]
Happy to be maintainer. Thanks!

**Comment by tlambert03** [comment]
@jakirkham @conda-forge/core, this is ready for review.  Thanks!

**Comment by wolfv** [comment]
You can still make the package noarch but prevent installation on macOS with a run dependency on the `__osx=*=thisdoesnotexist`

**Comment by xhochy** [comment]
> You can still make the package noarch but prevent installation on macOS with a run dependency on the `__osx=*=thisdoesnotexist`

It would be preferable if `noarch` packages that are noarch because of the unavailability of a dependency on a specific platform won't add any constraint. Rather, it is OK to merge here with failing CI on OSX or Win if it is due a missing dependency.

File: recipes/pycudadecon/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  version: {{ version }}
+
+source:
+  url: https://github.com/tlambert03/pycudadecon/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: c18a11c6536832ec4b69dc7ffad28acbc08b76af22df1a231353e05e429a6ff8
+
+build:
+  number: 0
+  script: {{ PYTHON }} -m pip install --no-deps --ignore-installed . -vv
+  skip: True  # [osx]
```
Comment (xhochy) [inline recipes/pycudadecon/meta.yaml:15]:
```suggestion
  script: {{ PYTHON }} -m pip install . -vv
  noarch: python
```

**Review by xhochy — APPROVED** [review]

**Comment by tlambert03** [comment]
thanks @xhochy.   I merged your commits, but now the OSX build is failing again as it did in 2644c69 ... can someone give me exact instructions on what the `__osx=*=thisdoesnotexist` line should look like?  I did a github [code search](https://github.com/search?p=1&q=filename%3Ameta.yaml+filename%3Ameta.yaml+%22__osx%22&type=Code) and can't seem to find any examples of people using that to _exclude_ OSX

**Comment by xhochy** [comment]
As said before: The OSX failure is fine :)

**Comment by tlambert03** [comment]
> The OSX failure is fine :)

ah, I see... it wasn't clear to me whether that "it would be preferable" statement was about conda forge in general, or my specific build.  It's definitely a bit jarring to have the big red X in the checks...

**Comment by xhochy** [comment]
This won't be a problem on the feedstock (no red crosses over there) and people trying to install this package on OSX will get a nice error message that `cudadecon` is not available for them. It's just not as nicely visible here on staged-recipes.

**Comment by tlambert03** [comment]
> This won't be a problem on the feedstock (no red crosses over there)

oh!  TIL :)  thanks for the help

**Pull request merged**